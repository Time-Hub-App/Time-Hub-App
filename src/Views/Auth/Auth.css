button {
  border-radius: 100px;
  background-color: transparent;
}
