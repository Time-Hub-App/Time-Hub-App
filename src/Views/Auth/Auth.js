import classNames from 'classnames';
import React, { useState } from 'react';
import AuthForm from '../../Components/AuthForm/AuthForm';
import { signInUser, signupUser } from '../../Services/users';


export default function Auth({ setUser }) {
  const [type, setType] = useState('Login');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [errorMsg, setErrorMsg] = useState('');
  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      let resp;
      if (type === 'Login') {
        resp = await signInUser(email, password);
      } else {
        resp = await signupUser(email, password);
      }
      setUser(resp);
    } catch {
      setErrorMsg('Your email/password is incorrect, try again or sign up. ');
    }
  };
  return (
    <div className="complete"><h3>{type}</h3><AuthForm email={email} setEmail={setEmail} password={password} setPassword={setPassword} errorMsg={errorMsg} handleSubmit={handleSubmit} />
      <button onClick={() => { setType('Sign Up'); } } className={classNames({ active: type === 'Sign Up' })}>Sign Up</button>
      <button onClick={() => { setType('Login'); } } className={classNames({ active: type === 'Login' })}>Login</button>
            
    </div>
  );
}
