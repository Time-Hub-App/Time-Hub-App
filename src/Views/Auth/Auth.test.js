import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Auth from './Auth';

it('should render auth view component', () => {
  const { container } = render(
    <MemoryRouter>
      <Auth />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
