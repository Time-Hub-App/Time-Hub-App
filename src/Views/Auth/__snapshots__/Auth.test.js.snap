// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render auth view component 1`] = `
<div>
  <div
    class="complete"
  >
    <header>
      <h3
        class=""
      >
        Home
      </h3>
      <h3
        class=""
      >
        About
      </h3>
      <h3
        class="active"
      >
        Log In
      </h3>
      <h3
        class=""
      >
        Journals
      </h3>
    </header>
    <h3>
      Login
    </h3>
    <form
      class="form"
    >
      <label>
        Email: 
      </label>
      <input
        type="email"
        value=""
      />
      <label>
        Password: 
      </label>
      <input
        type="password"
        value=""
      />
      <p />
      <button
        type="submit"
      >
        submit
      </button>
    </form>
    <button
      class=""
    >
      Sign Up
    </button>
    <button
      class="active"
    >
      Login
    </button>
    <footer>
      <a
        class="footer-link"
        href="https://github.com/Time-Hub-App"
      >
        TimeHub GitHub
      </a>
    </footer>
  </div>
</div>
`;
